### PR TITLE
Update contributing information

### PR DESCRIPTION
**Pull request by Louka3000**
- Move the GUI contributing information out from its directory and into the main CONTRIBUTING.md file.
- Clarify and add information
- Remove licenses in the GUI directory since GUI is part of the SlimeVR Server directory.

### CODE REVIEW

**Review by Vyolex — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,50 +1,66 @@
...
 
-## How to get started
+### Prerequisites
 
-### Getting the code
-First, clone the codebase using `git`. If you don't have `git` installed, go install it.
+- [Git](https://git-scm.com/downloads)
+- [Java v17+](https://adoptium.net/temurin/releases/)
+- [Node.js v16+](https://nodejs.org) (We recommend the use of `nvm` instead of installing Node.js directly)
+- [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section)
```
Comment (ImUrX) [inline CONTRIBUTING.md:10]: Aside from Windows now including Webview2 by default now, I don't see any mention of the usage of `webkit2gtk` on Linux

**Review by ImUrX — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (Louka3000) [reply CONTRIBUTING.md:10]: Can you propose changes. I’m not a Linux nerd.

Comment (ImUrX) [reply CONTRIBUTING.md:10]:
```suggestion
- [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section)
- [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section) or ``webkit2gtk`` for Linux
```

Comment (Louka3000) [reply CONTRIBUTING.md:10]: Is there a link or something that users should know about?

Comment (ImUrX) [reply CONTRIBUTING.md:10]: not worth it, linux is just kernel, depending on the distro they probably have different ways of installing :p

Comment (Louka3000) [reply CONTRIBUTING.md:10]: Oki

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,50 +1,66 @@
...
 
-## How to get started
+### Prerequisites
 
-### Getting the code
-First, clone the codebase using `git`. If you don't have `git` installed, go install it.
+- [Git](https://git-scm.com/downloads)
+- [Java v17+](https://adoptium.net/temurin/releases/)
+- [Node.js v16+](https://nodejs.org) (We recommend the use of `nvm` instead of installing Node.js directly)
+- [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section) or ``webkit2gtk`` for Linux
```
Comment (ButterscotchV) [inline CONTRIBUTING.md:10]:
```suggestion
- [Microsoft Edge WebView2](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section) or `webkit2gtk` for Linux
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,50 +1,66 @@
...
+dependencies.
+- You can run the server by running `./gradlew run` in your IDE's terminal.
+- To compile the code, run `./gradlew shadowJar`. The result will
+be at `server/build/libs/slimevr.jar` (you can ignore `server.jar`).
 
-### Building the code
-The code is built with `gradle`, a cli tool that manages java projects and their
-dependencies. You can build the code with `./gradlew build` and run it with
-`./gradlew run`.
+(Note: your IDE may do all of the above for you.)
```
Comment (ButterscotchV) [inline CONTRIBUTING.md:33]:
```suggestion
(Note: Your IDE may be able to do all of the above for you.)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,50 +1,66 @@
...
 
+### Tauri (gui)
 
-## Code Style
-Code is autoformatted with [spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle).
+- Run `npm install` in your IDE's terminal to download and install dependencies.
+- To launch the GUI in dev mode, run `npm run gui`.
```
Comment (ButterscotchV) [inline CONTRIBUTING.md:38]:
```suggestion
- To launch the GUI in dev mode, run `npm run tauri dev`.
```
Shouldn't this be like this instead? idk what `npm run gui` does.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,50 +1,66 @@
...
+### Java (server)
+
+The Java code is auto-formatted with [spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle).
 Code is checked for autoformatting whenever you build, but you can also run
 `./gradlew spotlessCheck` if you prefer.
 
-To autoformat your code from the command line, you can run `./gradlew spotlessApply`.
-We recommend installing support for spotless in your IDE of choice, and formatting
-whenever you save a file, to make things easy.
+To auto-format your Java code from the command line, you can run `./gradlew spotlessApply`.
```
Comment (ButterscotchV) [inline CONTRIBUTING.md:50]:
```suggestion
To auto-format your Java and Kotlin code from the command line, you can run `./gradlew spotlessApply`.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,50 +1,66 @@
...
+We recommend installing support for spotless in your IDE, and formatting
+whenever you save a file to make things easy.
 
-If you need to prevent autoformatting for a particular region of code, use
+If you need to prevent autoformatting for a select region of code, use
 `// @formatter:off` and `// @formatter:on`
 
-### Setting up spotless in VSCode
+#### Setting up spotless for IntelliJ IDEA
+* Install https://plugins.jetbrains.com/plugin/18321-spotless-gradle.
```
Comment (ButterscotchV) [inline CONTRIBUTING.md:58]:
```suggestion
* Install https://plugins.jetbrains.com/plugin/18321-spotless-gradle
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -73,19 +83,31 @@ Import the formatting settings defined in `spotless.xml`, like this:
...
-* You need to push this change with ``git push origin VERSION`` or ``git push origin --tags``
-(will push all tags you made).
+### Tauri (gui)
+
+We use ESLint and Prettier to format GUI code.
+- First, go into the GUI's directory with your terminal by running `cd gui`.
+- To check code formatting, run `npm run lint`.
+- To fix code formatting, run `npm run lint:fix` and `npm run format`
+
+Don't forget to run `cd ..` to go back in the main directory.
```
Comment (ButterscotchV) [inline CONTRIBUTING.md:93]:
```suggestion
Don't forget to run `cd ..` to return to the root directory.
```

File: README.md [hunk 8]
```diff
@@ -10,19 +10,23 @@ Sensors implementations:
 
 Integrations:
 * Use [SlimeVR OpenVR Driver](https://github.com/SlimeVR/SlimeVR-OpenVR-Driver) as a driver for SteamVR
```
Comment (ButterscotchV) [inline README.md:12]:
```suggestion
* Use [SlimeVR OpenVR Driver](https://github.com/SlimeVR/SlimeVR-OpenVR-Driver) as a driver for SteamVR.
```

**Review by ButterscotchV — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (Louka3000) [reply CONTRIBUTING.md:38]: `npm run gui` runs `npm i solarxr-protocol`, so it's better to avoid confusion and extra steps.

**Review by ButterscotchV — APPROVED** [review]

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**